packages/usdk/packages/upstreet-agent/packages/react-agents/lib/discord/discord-client.js

```diff
@@ -494,7 +514,32 @@ export class DiscordBotClient extends EventTarget {
     await readyPromise;
   }
 
+  async _handleReconnect(connectionParams) {
+    if (this.reconnectAttempts >= this.maxReconnectAttempts) {
```

Comment:
Seems like `reconnectAttempts` could be a local closure variable? If possible let's scope the state.